### DISCUSSION TITLE
Recommended way to transition from HACS to official integration?

### DISCUSSION THREAD

**Discussion by rivebe** [mletenay/home-assistant-goodwe-inverter#110] [Q&A] [answered]
First off, congratulations with having this great integration finally officially part of the latest Home Assistant release! Thank you very much for all the work and effort put into this project!

I for example have the goodwe integration running for a while now via HACS. But which steps do I follow now if I want to use the official integration using the default Home Assistant appraoch? Should I first uninstall the current HACS integration? Or can I install it right next to it. What with the sensors, will I have duplicate ones using the same names?

Lot's of questions, and I want to be informed as well as I can before I move from one integration to the other goodwe integration.

Thank you for any advice you have to share on this transition @mletenay!

**Comment by mletenay** [comment, marked as answer]
I will publish migration information soon, but meanwhile:
If you want to switch from HACS to HA and keep all your entity names, history, statistics etc., do NOT uninstall the integration from HA.
(The HA version of the integration has different default names for sensors).
Just go to HACS and uninstall the custom component - it will complain it is used - just press Ignore and uninstall the HACS version anyway.
After restart of HA, the standard integration will start right away, keeping all the data from HACS version.
(With usual recommendation - backup first, in case something goes wrong).

**Reply by rivebe** [reply]
Thank you, I’ll wait for your official migration information to be sure I do it correctly. Where can we expect it to appear exactly? So I can monitor it…

From what I understood here, but don’t dare to do:
1) I should uninstall custom (currently running/working) goodwe integration from HACS, and I would not loose any data.
2) I should follow step 1 then by installing the official goodwe integration in HA
… but I’m affraid since the sensor names will have changed, I’ll have to change it on all the places where I had the old sensor names defined in HA.

**Reply by rivebe** [reply]
Apologies, I just noticed it here: https://github.com/mletenay/home-assistant-goodwe-inverter but I still wonder. 

So first install/configure official HA goodwe integration? …. then uninstall HACS goodwe integration, and after restart all should be fine?

**Reply by mletenay** [reply]
Yes, resp. no ;-)
There's no special HA goodwe integration installation.
Just upgrade your HA to 2022.2.x and then remove the HACS component.

**Comment by LubosKovar** [comment]
Hi, I have oposit question. Becasue I need change inverter mode, custom HA integration does not supply this function. I need migrate to HACS experimental version. What is the best migration process to safe my data from original HA-GW integration to HACS-GW?
